mypyc/test-data/irbuild-classes.test

```diff
@@ -1607,3 +1607,16 @@ def native_class(x):
 L0:
     r0 = CPy_TYPE(x)
     return r0
+
+[case testUnsupportedDunderNew]
```

Comment:
Also add a successful irbuild test that calls `super().__new__(cls)` in a `__new__` method. A simple case is sufficient (i.e. don't return a union type from `__new__`).